src/ol_infrastructure/applications/redash/__main__.py

```diff
@@ -85,10 +117,39 @@
     path="/ol-data/redash-profile/",
 )
 
+redash_server_vault_policy = vault.Policy(
+    "redash-server-vault-policy",
+    name="redash-server",
+    policy=Path(__file__).parent.joinpath("redash_server_policy.hcl").read_text(),
+)
+vault.aws.AuthBackendRole(
+    "redash-server-ami-ec2-vault-auth",
+    backend="aws",
+    auth_type="iam",
+    role="redash-server",
+    inferred_entity_type="ec2_instance",
+    inferred_aws_region=aws_config.region,
+    bound_iam_instance_profile_arns=[redash_instance_profile.arn],
+    bound_ami_ids=[redash_ami.id],
+    bound_account_ids=[aws_account.account_id],
+    bound_vpc_ids=[data_vpc["id"]],
+    token_policies=[redash_server_vault_policy.name],
+)
+
 redash_instance_security_group = ec2.SecurityGroup(
     f"redash-instance-{stack_info.env_suffix}",
     description="Security group to assign to Redash application to control "
     "inter-service access",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
```

Comment:
Rather than creating a specific security group for allowing web traffic we can just use the pre-made `web` security group from the network stack output. We may also want to assign the `data-integrator` security group for allowing access to the various databases until we sever those connections and move things into the lake/dbt workflow.

Reply:
Ummm, well that isn't the only thing this group is used for. I think maybe it is just convenient and explicit to include the HTTPS port here. It is used as an ingress for redash db security group, lines 158-187 and then in a similar manner for redis on lines 238-255.

If I were to use the default web group as an ingress target for this, then I would be opening up any web server to talk to Postgres + Redis, which I don't think we necessarily want, right? 

👍 on the data-integrator thing tho.

Reply:
All valid points :+1: